### ISSUE TITLE
CS8019 doesn't work during compilation

### ISSUE THREAD

**Issue by msallin** [dotnet/roslyn#41391]
**Version Used**: 
- dotnet --version: 3.1.100
- VS Enterprise 16.4.0

**Steps to Reproduce**:
1. Create new Class Library (.NET Framework) project.
2. Edit Active Rule Set and make CS8019 (Unnecessary using directive) an 'Error'.
3. Compile project and/or run "dotnet build"

**Expected Behavior**:
Given that the project starts with a Class1.cs file containing unused using directives, compilation should fail.

**Actual Behavior**:
Compilation succeeds. The VS UI does show the IntelliSense error, however. dotnet build shows no warnings and no errors.

[ConsoleApp4.zip](https://github.com/dotnet/roslyn/files/4154792/ConsoleApp4.zip)

**Comment by sharwell** [comment]
@dotnet/roslyn-compiler I saw that reporting these diagnostics is tied to the `DocumentationMode`:

https://github.com/dotnet/roslyn/blob/69d3fb733e6c74a41c118bf905739163cf5aef2a/src/Compilers/CSharp/Portable/Compilation/CSharpCompilation.cs#L2124-L2134

However, even with `GenerateDocumentationFile=true` I was not able to observe CS8019 diagnostics for command line builds.

**Comment by sharwell** [comment]
I investigated this issue, and found that it is by design. Since CS8019 does not filter generated code, it's essentially impossible to increase the severity of this diagnostic and still have a build that works. For example, the **{project}.AssemblyInfo.cs** files that are generated by default for all SDK projects have unused using directives.

**Comment by msallin** [comment]
Hi Sam

Thank you for your quick response. Any idea of a workaround or a replacemenet to get unused usings reported like I expect it by CS8019?

Bascially, I want to let a build fail if there are unused usings.

Thanks, Marc

**Comment by sharwell** [comment]
Currently no such analyzer is provided, but @mavasani is working to make IDE0005 available in command line scenarios.

**Comment by ericburcham** [comment]
I don't have time to wait for a public Analyzer.  I need to fail the build during compilation for unneeded using statements now.  Where would I go to look at the source code for CS8019 so we can make our own internal version?

**Comment by gafter** [comment]
@ericburcham Look in `ErrorCode.cs` for which `enum` value is assigned the value 8019.  Then look to see where that enum value is used.

**Comment by mavasani** [comment]
@ericburcham The IDE0005 analyzer for remove unnecessary usings is now part of this NuGet package: https://dotnet.myget.org/feed/roslyn/package/nuget/Microsoft.CodeAnalysis.CSharp.CodeStyle/3.6.0-2.20157.5

You can enforce this on command line for your project as follows:
1. Install the NuGet package to your project
2. Mark it as a build warning in your repo level editorconfig/ruleset file:
```ini
[*.{cs,vb}]

# IDE0005: Remove unnecessary usings/imports
dotnet_diagnostic.IDE0005.severity = warning
```
3. Enable XML documentation comments for your project. This is needed to workaround https://github.com/dotnet/roslyn/issues/41640

**Comment by ericburcham** [comment]
Sweeeet!

We have an analyzer working, but it will be nice to remove the custom one.

Sent from a mobile device.  Please excuse errors / brevity.

> On Mar 7, 2020, at 10:29 AM, Manish Vasani <notifications@github.com> wrote:
> 

﻿
@ericburcham The IDE0005 analyzer for remove unnecessary usings is now part of this NuGet package: https://dotnet.myget.org/feed/roslyn/package/nuget/Microsoft.CodeAnalysis.CSharp.CodeStyle/3.6.0-2.20157.5

You can enforce this on command line for your project as follows:

Install the NuGet package to your project
Mark it as a build warning in your repo level editorconfig/ruleset file:
[*.{cs,vb}]

# IDE0005: Remove unnecessary usings/imports
dotnet_diagnostic.IDE0005.severity = warning
Enable XML documentation comments for your project. This is needed to workaround #41640

**Comment by cezarypiatekGC** [comment]
@mavasani Are you going to release `Microsoft.CodeAnalysis.CSharp.CodeStyle` package via `nuget.org` ? I'm not able to use myget as a nuget feed inside my company.
